q2_american_gut/_fetch.py

```diff
@@ -0,0 +1,205 @@
+    """Fetch phylogeny based on the processing type
+
+    Parameters
+    ----------
+    processing_type : {'deblur', 'closed-reference'}
+        The processing type which determines how we resolve a phylogeny.
+    table : biom.Table
+        The FrequencyTable
+    threads : int
+        The number of threads to use
+    debug : bool, optional
+        If debug, use a small small tree for insertion to avoid spinup
+        time.
+
+    Returns
+    -------
+    Phylogeny[Rooted]
+    """
+    if processing_type == 'deblur':
+        from qiime2.plugins.fragment_insertion.methods import sepp
+
+        # insertion is extremely expensive to spin up (20min) so short
+        # circuit if debug
+        if debug:
+            ref_phy = qiime2.Artifact.load(resource_filename(*DEBUG_PHY))
+            #ref_phy = ref_phy.view(NewickFormat)
+
+            ref_aln = qiime2.Artifact.load(resource_filename(*DEBUG_ALN))
+            #ref_aln = ref_aln.view(AlignedDNASequencesDirectoryFormat)
+        else:
+            ref_phy = None
+            ref_aln = None
+
+        dna_iter = _get_featuredata_from_table(table)
+        reads = qiime2.Artifact.import_data('FeatureData[Sequence]', dna_iter)
+        tree, _ = sepp(reads, threads=threads,
+                       reference_alignment=ref_aln,
+                       reference_phylogeny=ref_phy)
+
+    else:
+        tree = skbio.TreeNode.read(resource_filename(*GG_TREE))
+        tree = qiime2.Artifact.import_data('Phylogeny[Rooted]', tree)
+
+    return tree
+
+
+def fetch_amplicon(qiita_study_id: str, processing_type: str, trim_length: int,
+                   threads: int=1, debug: bool=False) -> (biom.Table,
+                                                          pd.DataFrame,
+                                                          pd.DataFrame,
+                                                          skbio.TreeNode):
+    if not qiita_study_id.isdigit():
+        raise ValueError("ID %s is not a qiita ID" % qiita_study_id)
+
+    if int(qiita_study_id) < 1:
+        raise ValueError("ID %s is not a qiita ID" % qiita_study_id)
+
+    context = _determine_context(processing_type, trim_length)
+
+    query = "where qiita_study_id==%s" % qiita_study_id
+    samples = redbiom.search.metadata_full(query)
+
+    if not samples:
+        raise ValueError("study ID %s has no samples or is not a Qiita ID")
+
+    if debug:
+        # just take top 10 but be consistent about which 10 are obtained
+        samples = set(sorted(samples)[:10])
+
+    table, ambiguity_map_tab = redbiom.fetch.data_from_samples(context, samples)
+    md, ambiguity_map_md = redbiom.fetch.sample_metadata(samples, context=context)
+    md.set_index('#SampleID', inplace=True)
+
+    taxonomy = _fetch_taxonomy(processing_type, table, threads)
+    phylogeny = _fetch_phylogeny(processing_type, table, threads, debug)
+
+    md = qiime2.Artifact.import_data(QiitaMetadata, md)
+    md = qiime2.Metadata.from_artifact(md)
```

Comment:
This function is no longer supported in qiime2-2018.2. It should be replaced with `Artifact.view(Metadata)`. See more at the latest [qiime2 docs](https://forum.qiime2.org/t/summary-of-changes-to-metadata-in-qiime-2-2018-2-release/2884)